### ISSUE TITLE
Eval without parsing the source

### ISSUE THREAD

**Issue by SchrodingerZhu** [oracle/graal#5598]
Hi,

I know that Call Target can be invoked by using Context::eval. But sometimes I may already have the AST in Java. Can AST be directly evaluated without serializing into source code?

**Comment by oubidar-Abderrahim** [comment]
Hi, thank you for reaching out, could you please share a code example illustrating what you're asking? also, specify the expected behavior and the actual one. Thank you

**Comment by chumer** [comment]
Not really. I'd definitely not recommend doing that. I'd recommend creating the AST within Truffle. e.g. in TruffleLanguage.parse from some source or binary representation. 

You can pass object into the language that is not necessarily a string or binary representation. E.g. something backed by a Java host object. You can convert that to a guest language AST by interpreting the alternative representation using Truffle interop. Reusing the same Java AST objects for that is something you can only do if you break the polyglot/Truffle encapsulation. 

Breaking polygot/Truffle encapsulation is something you should only do, if you don't care at all about any other truffle languages or tooling, because those might not work. You might also run some into assertions when you try to use Truffle context thread locals. But if you want to force the breakage you can disable class loading separation [1] and pass the AST in e.g. with a ThreadLocal (set it, and then access it in TruffleLanguage.parse). Again, avoid this path if you can and only use it for testing, which we sometimes do in the Truffle / language tests.

[1] https://www.graalvm.org/latest/graalvm-as-a-platform/implement-language/#run-simplelanguage-using-command-line see section Disable Class Path Separation.
